Add Duplicity package

adrien-delhorme opened the pull request:
Is there someone that can review my PR? This is my first contribution to spksrc.
I only tried to cross compile for the arch-88f6281.
Any help appreciated.

mnonnenmacher commented:
I can verify that the package works on my DS214 (arch-armadaxp), when I also install gpg-agent. Great work :+1:

Dr-Bean commented on cross/duplicity/Makefile:
```diff
@@ -1,15 +1,15 @@
 PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
 
-DEPENDS = cross/librsync
+DEPENDS = cross/librsync cross/netifaces cross/wrapt
```
`cross/netifaces` and `cross/wrapt` should be removed here, as well as removed from `spk/Python/Makefile`. Both entries should be added to BUILD_DEPENDS of `spk/duplicity/Makefile`

Dr-Bean commented on cross/duplicity/Makefile:
```diff
@@ -1,15 +1,15 @@
 HOMEPAGE = http://duplicity.nongnu.org/
 COMMENT  = Encrypted bandwidth-efficient backup using the rsync algorithm
-LICENSE  =
+LICENSE  = GNU GPLv2 
```
nit: unnecessary space ;)

Dr-Bean commented on cross/librsync/Makefile:
```diff
@@ -12,6 +12,6 @@ COMMENT  = librsync is a free software library that implements the rsync remote-
 LICENSE  =
 
 GNU_CONFIGURE = 1
-CONFIGURE_ARGS = --disable-static --enable-shared
```
Personally, I'd prefer to have `librsync` shared. So, instead of doing this, the alternative is to keep this line, keep the PLIST, add `DEPENDS=cross/librsync` to the spk's Makefile, and remove `COPY_TARGET = nop` from there.

Dr-Bean commented on cross/openssl/digests.orig:
```diff
@@ -0,0 +1,9 @@
+<<<<<<< HEAD
```
This needs fixing. Don't `git pull` in a branch you use for PR's, unless you also rebase.

Dr-Bean commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,43 @@
+SPK_VERS = 0.7.03
+SPK_REV = 1
+SPK_ICON = src/duplicity.png
+DSM_UI_DIR = app
```
There's no UI, so this can be dropped

Dr-Bean commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,43 @@
+	install -m 755 -d $(STAGING_DIR)/share
+	install -m 755 -d $(STAGING_DIR)/share/wheelhouse
+	install -m 644 $(WORK_DIR)/wheelhouse/* $(STAGING_DIR)/share/wheelhouse/
+	for size in 16 24 32 48 72 ; \
```
There's no UI, so this loop can be dropped, as well as L34-35.

Dr-Bean commented on spk/duplicity/src/requirements.txt:
```diff
@@ -0,0 +1,21 @@
+iso8601==0.1.10
+lockfile==0.10.2
+netaddr==0.7.14
+netifaces==0.10.4
```
Suggestion: remove the versions for wheels that you create (and move to top or bottom of the file, to distinguish them easier, and you can even add a comment). Othrwise, when the cross package is updated and this requirements file is overlooked, the package won't compile anymore.

Dr-Bean commented on spk/python/Makefile:
```diff
@@ -11,7 +11,8 @@ DEPENDS += cross/pyopenssl cross/pycrypto cross/m2crypto
 DEPENDS += cross/sqlalchemy cross/pyaudio cross/pyalsa cross/lxml cross/twisted
 DEPENDS += cross/zope.interface cross/markupsafe cross/sphinxbase cross/pocketsphinx
-DEPENDS += cross/pyyaml cross/msgpack-python cross/pymongo cross/uwsgi
+DEPENDS += cross/pyyaml cross/msgpack-python cross/pymongo cross/uwsgi cross/wrapt
```
No need to add this or L15 here, see my first comment.

Dr-Bean commented:
Nice work :) Added some comments.

Dr-Bean commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,43 @@
+
+MAINTAINER = SynoCommunity
+DESCRIPTION = Encrypted bandwidth-efficient backup using the rsync algorithm.
+RELOAD_UI = yes
```
If there's no UI, and package is not startable via GUI (i.e. command line only), there's no need for this.

adrien-delhorme commented:
Thanks for your reviews! I will fix the PR in few days.

adrien-delhorme commented:
I fixed my branch to remove the `git pull` of master branch. And I incorporated all the comments you made. Let me know if there are other things to change.

Dr-Bean commented on cross/boto/Makefile:
```diff
@@ -0,0 +1,15 @@
+PKG_NAME = boto
+PKG_SHORT_VERS = 2.38
```
You don't seem to need `PKG_SHORT_VERS` anywhere for Boto. Please just refer to `PKG_VERS=2.38.0` if that's correct.

Dr-Bean commented on cross/boto/Makefile:
```diff
@@ -0,0 +1,15 @@
+PKG_DIST_SITE = https://github.com/$(PKG_NAME)/$(PKG_NAME)/archive
+PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
+
+#DEPENDS = cross/librsync
```
Please put an empty `DEPENDS` here if librsync is not a dependency.

Dr-Bean commented on cross/duplicity/Makefile:
```diff
@@ -1,15 +1,15 @@
 PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
 
-DEPENDS = cross/librsync
+DEPENDS = cross/boto cross/librsync
```
`cross/boto` is a wheel. So, it belongs in `spk/duplicity/Makefile` as `BUILD_DEPENDS`, not as a dependency of another wheel.

Dr-Bean commented on cross/duplicity/patches/setup.py.patch:
```diff
@@ -0,0 +1,63 @@
+     print "Sorry, duplicity requires version 2.4 or later of python"
+     sys.exit(1)
+ 
++if sys.version_info[0] >= 3:
```
I don't see how this part of the patch is necessary, seeing as we're only interested in Python 2.7 support. Please verify this is needed.

Dr-Bean commented on cross/librsync/Makefile:
```diff
@@ -15,3 +15,4 @@ GNU_CONFIGURE = 1
 CONFIGURE_ARGS = --disable-static --enable-shared
 
 include ../../mk/spksrc.cross-cc.mk
+
```
Don't change existing files please :)

Dr-Bean commented on cross/librsync/PLIST:
```diff
@@ -2,4 +2,3 @@ bin:bin/rdiff
 lib:lib/librsync.so.1.0.2
 lnk:lib/librsync.so.1
 lnk:lib/librsync.so
-
```
Same here.

Dr-Bean commented on cross/openssl/Makefile:
```diff
@@ -1,5 +1,5 @@
 PKG_NAME = openssl
-PKG_VERS = 1.0.2a
+PKG_VERS = 1.0.2b
```
This PR is going to need a rebase: we're already on 1.0.2d.

Dr-Bean commented on cross/openssl/digests:
```diff
@@ -1,3 +1,3 @@
-openssl-1.0.2a.tar.gz SHA1 46ecd325b8e587fa491f6bb02ad4a9fb9f382f5f
-openssl-1.0.2a.tar.gz SHA256 15b6393c20030aab02c8e2fe0243cb1d1d18062f6c095d67bca91871dc7f324a
-openssl-1.0.2a.tar.gz MD5 a06c547dac9044161a477211049f60ef
+openssl-1.0.2b.tar.gz SHA1 9006e53ca56a14d041e3875320eedfa63d82aba7
```
See previous comment.

Dr-Bean commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,34 @@
+SPK_ICON = src/duplicity.png
+
+DEPENDS = cross/librsync
+BUILD_DEPENDS = cross/python cross/setuptools cross/pip cross/wheel cross/netifaces cross/wrapt cross/$(SPK_NAME)
```
- Nit: BUILD_DEPENDS comes before DEPENDS
- Line too long. Please put a line break somewhere in here, so you end up with something like: 
```
BUILD_DEPENDS = cross/python cross/setuptools cross/pip cross/wheel 
BUILD_DEPENDS += cross/boto cross/netifaces cross/wrapt cross/$(SPK_NAME)
```
(`cross/boto` added, because this is where it needs to go anyway ;))

Dr-Bean commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,34 @@
+DEPENDS = cross/librsync
+BUILD_DEPENDS = cross/python cross/setuptools cross/pip cross/wheel cross/netifaces cross/wrapt cross/$(SPK_NAME)
+WHEELS = src/requirements.txt
+SPK_DEPENDS = "python>=2.7:gnupg>=2.1"
```
Use the full version. `python>=2.7.8-9` for wheel support. Same for gnupg, which would probably be `gnupg>=2.1-1`

Dr-Bean commented on spk/duplicity/src/installer.sh:
```diff
@@ -0,0 +1,60 @@
+    ln -s ${SYNOPKG_PKGDEST} ${INSTALL_DIR}
+
+    # Create a Python virtualenv
+    ${VIRTUALENV} --system-site-packages ${INSTALL_DIR}/env > /dev/null
```
I think you can get away with removing `--system-site-packages`. In doing that, the package is going to be easier to maintain in the long run. Could you check if that works?

Dr-Bean commented:
Added some more comments. Due to a browser crash, I had to start over, but I think I've reproduced all the comments.
As I said, if seem this PR has conflicts, probably because of the OpenSSL changes. Could you rebase on top of master please?

Dr-Bean commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,34 @@
+SPK_REV = 1
+SPK_ICON = src/duplicity.png
+
+DEPENDS = cross/librsync
```
Missed this one: I don't think you need this here, please check.

zaninime commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,34 @@
+SPK_NAME = duplicity
+SPK_VERS = 0.7.03
```
Version 0.7.05 is out: http://duplicity.nongnu.org/index.html
Just to keep the package "bleeding edge" :)

Randur commented:
Any progress on this?
Having duplicity package would be great!

adrien-delhorme commented:
I'm currently working on the GnuPG package that is a dependency of Duplicity. I will probably finish and update the pull request next week.

adrien-delhorme commented:
I updated en rebased the pull request.

Dr-Bean commented on cross/boto/Makefile:
```diff
@@ -0,0 +1,14 @@
+PKG_VERS = 2.38.0
+PKG_EXT = tar.gz
+PKG_DIST_NAME = $(PKG_VERS).$(PKG_EXT)
+PKG_DIST_SITE = https://github.com/$(PKG_NAME)/$(PKG_NAME)/archive
```
PyPI the preferred location for Python projects if they're available there. Download Boto from here: https://pypi.python.org/packages/source/b/boto/boto-2.38.0.tar.gz.

Dr-Bean commented on cross/duplicity/patches/setup.py.patch:
```diff
@@ -0,0 +1,45 @@
+                 'install': InstallCommand,
+-                'sdist': SDistCommand},
++                'sdist': SDistCommand,
++                'install_scripts': InstallScriptsCommand,},
```
Minor: there's a comma which doesn't seem to belong here ;)

Dr-Bean commented on cross/librsync/Makefile:
```diff
@@ -14,4 +14,4 @@ LICENSE  =
 CONFIGURE_ARGS = --disable-static --enable-shared
 
-include ../../mk/spksrc.cross-cc.mk
+include ../../mk/spksrc.cross-cc.mk
```
Revert this file to original please. Remove any and all changes to it.

Dr-Bean commented on cross/librsync/PLIST:
```diff
@@ -1,5 +1,4 @@
 lnk:lib/librsync.so.1
-lnk:lib/librsync.so
-
+lnk:lib/librsync.so
```
Revert this file to original please. Remove any and all changes to it.

Dr-Bean commented on spk/duplicity/Makefile:
```diff
@@ -0,0 +1,35 @@
+DESCRIPTION = Encrypted bandwidth-efficient backup using the rsync algorithm.
+STARTABLE = no
+DISPLAY_NAME = Duplicity
+CHANGELOG = ""
```
Could be set to `Initial release`.

Dr-Bean commented:
You might want to rebase/redo commits, and set a correct author while you're at it (you'll notice that your current first two commits are on an unassociated author name, or whatever git calls them).

If nothing else changes but the last couple of commits, I believe this should be good to go. Thanks so far :)

adrien-delhorme commented:
I think it's ok.
Sorry for the "randomly" changing files ^^. I'm a newbie with git and Github: this is my first real pull request.
Thank you for the review.

Dr-Bean commented:
Fyi, there are a couple things that need correcting (boto digests, missing librsync lib in the SPK). I'm doing that as a set of extra commits while merging and building the packages, so we can get the packages published today.

Dr-Bean commented:
duplicity 0.7.05-1 has been published on the repository as a beta package. 
Additional changes on top of your PR: https://github.com/SynoCommunity/spksrc/commit/99a75ddd9ba106a48cfe3caedd35cd29e78f8c32 and https://github.com/SynoCommunity/spksrc/commit/bf28e2a09cf57f1faef7675c9d2553e054a2277a.
